source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -265,18 +267,18 @@ Network::IoResult SslSocket::doWrite(Buffer::Instance& write_buffer, bool end_st
   return {PostIoAction::KeepOpen, total_bytes_written, false};
 }
 
-void SslSocket::onConnected() { ASSERT(state_ == Ssl::SocketState::PreHandshake); }
+void SslSocket::onConnected() { ASSERT(info_->state() == Ssl::SocketState::PreHandshake); }
 
 Ssl::ConnectionInfoSharedPtr SslSocket::ssl() const { return info_; }
 
 void SslSocket::shutdownSsl() {
-  ASSERT(state_ != Ssl::SocketState::PreHandshake);
-  if (state_ != Ssl::SocketState::ShutdownSent &&
+  ASSERT(info_->state() != Ssl::SocketState::PreHandshake);
+  if (info_->state() != Ssl::SocketState::ShutdownSent &&
       callbacks_->connection().state() != Network::Connection::State::Closed) {
     int rc = SSL_shutdown(rawSsl());
     ENVOY_CONN_LOG(debug, "SSL shutdown: rc={}", callbacks_->connection(), rc);
     drainErrorQueue();
-    state_ = Ssl::SocketState::ShutdownSent;
+    info_->state() = Ssl::SocketState::ShutdownSent;
```

Comment:
Worth adding a separate mutation method and have state() return the state by value?

Reply:
Done.